### ISSUE TITLE
Warn on collection membership checks that resolve to Never

### ISSUE THREAD

**Issue by dmvinson** [microsoft/pyright#5218]
**Is your feature request related to a problem? Please describe.**
Currently, membership checks that resolve to Never don't produce a warning, but this is almost always a mistake. An example is:
```
a: set[int] = {1, 2, 3}
b: set[str] = {"a", "b", "c"}

for item in a:
    if item in b:
        print(item) # type here is Never
```

**Describe the solution you'd like**
This membership check will resolve to Never (always False), it'd be nice if there was an optional configuration to make this a warning diagnostic, similar to reportUnnecessaryComparison.

**Comment by erictraut** [comment]
You probably already realized this, but pyright already has a diagnostic check for this condition: `reportUnnecessaryContains`. It's not enabled by default in "basic" type checking mode, but you can enable it.

**Comment by dmvinson** [comment]
Yes, thank you! Was just testing this out after finding it
